src/lib/mdns/Advertiser_ImplMinimalMdns.cpp

```diff
@@ -165,77 +169,129 @@ class AllInterfaces : public mdns::Minimal::ListenIterator
 };
 
 class AdvertiserMinMdns : public ServiceAdvertiser,
-                          public mdns::Minimal::ServerDelegate, // gets queries
-                          public mdns::Minimal::ParserDelegate  // parses queries
+                          public ServerDelegate, // gets queries
+                          public ParserDelegate  // parses queries
 {
 public:
-    AdvertiserMinMdns() :
-        mResponseSender(&mServer, &mQueryResponder), mPtrResponder(mOperationalServiceQName, mOperationalServerQName),
-        mSrvResponder(mOperationalServerQName, mdns::Minimal::SrvResourceRecord(mOperationalServerQName, mServerQName, CHIP_PORT)),
-        mIPv4Responder(mServerQName), mIPv6Responder(mServerQName),
-        mTxtResponder(mdns::Minimal::TxtResourceRecord(mOperationalServerQName, mEmptyTextEntries))
+    AdvertiserMinMdns() : mResponseSender(&mServer, &mQueryResponder)
 
     {
         mServer.SetDelegate(this);
+
+        for (size_t i = 0; i < kMaxAllocatedResponders; i++)
+        {
+            mAllocatedResponders[i] = nullptr;
+        }
+        for (size_t i = 0; i < kMaxAllocatedQNameData; i++)
+        {
+            mAllocatedQNameParts[i] = nullptr;
+        }
     }
+    ~AdvertiserMinMdns() { Clear(); }
 
     // Service advertiser
     CHIP_ERROR Start(chip::Inet::InetLayer * inetLayer, uint16_t port) override;
     CHIP_ERROR Advertise(const OperationalAdvertisingParameters & params) override;
 
     // ServerDelegate
-    void OnQuery(const mdns::Minimal::BytesRange & data, const chip::Inet::IPPacketInfo * info) override;
-    void OnResponse(const mdns::Minimal::BytesRange & data, const chip::Inet::IPPacketInfo * info) override {}
+    void OnQuery(const BytesRange & data, const chip::Inet::IPPacketInfo * info) override;
+    void OnResponse(const BytesRange & data, const chip::Inet::IPPacketInfo * info) override {}
 
     // ParserDelegate
-    void OnHeader(mdns::Minimal::ConstHeaderRef & header) override { mMessageId = header.GetMessageId(); }
-    void OnResource(mdns::Minimal::ResourceType type, const mdns::Minimal::ResourceData & data) override {}
-    void OnQuery(const mdns::Minimal::QueryData & data) override;
+    void OnHeader(ConstHeaderRef & header) override { mMessageId = header.GetMessageId(); }
+    void OnResource(ResourceType type, const ResourceData & data) override {}
+    void OnQuery(const QueryData & data) override;
 
 private:
-    static constexpr size_t kMaxEndPoints = 10;
-    static constexpr size_t kMaxRecords   = 16;
+    /// Sets the query responder to a blank state and frees up any
+    /// allocated memory.
+    void Clear();
+
+    /// Append another responder to the internal replies
```

Comment:
super nit: These comments keep switching mood.

"Set the query responder to a blank state and free up any allocated memory."

"Appends another responder to the internal replies."

[Note Google style is to always use the latter, like you've done for Clear]